File: backend/src/main/java/com/hyunprise/backend/domain/coupon/vo/IssuedCoupon.java
package com.hyunprise.backend.domain.coupon.vo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class IssuedCoupon {
    private String issuedCouponUUID;
    private String memberUUID;
    private String couponUUID;
    private String couponCode;
    private Integer status;
    private Timestamp issueDate;
    private Timestamp expirationDate;
    private Integer membershipPoint;
}

File: android/app/src/main/java/com/hyunprise/android/api/coupon/vo/IssuedCoupon.kt
package com.hyunprise.android.api.coupon.vo

import java.sql.Timestamp


data class IssuedCoupon(
    var issuedCouponUUID: String,
    var memberUUID: String,
    var couponUUID: String,
    val couponCode: String = "defaultCode",
    val status: Int = 0,
    val issueDate: Timestamp? = null,
    val expirationDate: Timestamp?= null
) {
    constructor(memberUUID: String, couponUUID: String)
            : this(
        issuedCouponUUID="defaultCode",
        memberUUID = memberUUID,
        couponUUID = couponUUID
    )
}

